compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/internal/ClipboardUtils.desktop.kt

```diff
@@ -63,6 +85,32 @@ internal actual fun ClipEntry?.hasText(): Boolean {
 // Note: the name can't be just `hasText` because NativeClipboard is a typealias to Any,
 // so it would conflict with ClipEntry?.hasText declaration. Therefore, we need a unique name.
 internal fun NativeClipboard.nativeClipboardHasText(): Boolean {
-    val awtClipboard = this as? java.awt.datatransfer.Clipboard ?: return false
+    val awtClipboard = this as? Clipboard ?: return false
     return awtClipboard.isDataFlavorAvailable(DataFlavor.stringFlavor)
-}
\ No newline at end of file
+}
+
+// Derived from StringSelection
+@VisibleForTesting
+internal class AnnotatedStringTransferable(
+    private val data: AnnotatedString
+) : Transferable, ClipboardOwner {
+    override fun getTransferDataFlavors(): Array<DataFlavor?> = supportedFlavors
+
+    override fun isDataFlavorSupported(flavor: DataFlavor): Boolean =
+        flavor in supportedFlavors
+
+    override fun getTransferData(flavor: DataFlavor): Any =
+        when (flavor) {
+            _annotatedStringFlavor -> data
+            DataFlavor.stringFlavor -> data.text
+            else -> throw UnsupportedFlavorException(flavor)
+        }
+
+    override fun lostOwnership(clipboard: Clipboard?, contents: Transferable?) {
+        // Empty
```

Comment:
I don't know why `StringSelection` implements `ClipboardOwner` only to have its only function do nothing; I however am going to retain the same behaviour, just in case it covers some unintuitive edge cases. It shouldn't cause any issues, judging by the JavaDoc of `ClipboardOwner`.